### PR TITLE
List item component

### PR DESCRIPTION
**Pull request by eldiegod**
fixes: #1635

https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=3152%3A103472

### CODE REVIEW

File: src/components/ListItem/ListItem.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    disabled: false,
+    destructive: false,
+  },
+} as Meta
+
+const Template: Story<ListItemProps> = (args) => (
+  <Container>
+    <ListItem {...args} />
+    <ListItem {...args} />
+    <ListItem {...args} />
```
Comment (drillprop) [inline src/components/ListItem/ListItem.stories.tsx:35]: Could you also add a template with a single `ListItem` to see more clearly how they look in isolation?

File: src/components/ListItem/ListItem.styles.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      return css`
+        padding: ${sizes(2)} ${sizes(4)};
+      `
+    case 'small':
+      return css`
+        padding: ${sizes(1)} ${sizes(4)};
+      `
+  }
+}
+type ContainerProps = { size: ListItemSizes; hasNodeStart: boolean; disabled?: boolean }
```
Comment (drillprop) [inline src/components/ListItem/ListItem.styles.ts]: Let's move this close to the Container
```suggestion
```

File: src/components/ListItem/ListItem.styles.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  &:focus,
+  &:hover {
+    background-color: ${cVar('colorBackgroundAlpha')};
+  }
+
+  &:active {
+    background-color: ${cVar('colorBackgroundMutedAlpha')};
+  }
+`
+export const Container = styled.div<ContainerProps>`
```
Comment (drillprop) [inline src/components/ListItem/ListItem.styles.ts:41]:
```suggestion
type ContainerProps = { size: ListItemSizes; hasNodeStart: boolean; disabled?: boolean }
export const Container = styled.div<ContainerProps>`
```

File: src/components/ListItem/ListItem.styles.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      return css`
+        padding: ${sizes(1)} ${sizes(4)};
+      `
+  }
+}
+type ContainerProps = { size: ListItemSizes; hasNodeStart: boolean; disabled?: boolean }
+const disabledStyles = css`
+  opacity: 0.25;
+  cursor: not-allowed;
+`
```
Comment (drillprop) [inline src/components/ListItem/ListItem.styles.ts:29]: hmm I think disabled variants looks different in the designs: 
image

File: src/components/ListItem/ListItem.styles.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    background-color: ${cVar('colorBackgroundMutedAlpha')};
+  }
+`
+export const Container = styled.div<ContainerProps>`
+  display: grid;
+  grid-gap: ${sizes(3)};
+  grid-auto-flow: column;
+  grid-template-columns: ${({ hasNodeStart }) => (hasNodeStart ? 'auto 1fr' : '1fr')};
+  align-items: center;
+  user-select: none;
```
Comment (drillprop) [inline src/components/ListItem/ListItem.styles.ts:47]: This component should have also different default background if Im not mistaken. 
```
background: var(--color-background-muted);
```
Design: 
image

File: src/components/Text/Text.styles.ts [hunk 6]
```diff
@@ -4,12 +4,13 @@ import styled from '@emotion/styled'
...
 
 type TextProps = {
+  color?: string
   isSecondary?: boolean
 }
 
 const baseStyles = (props: TextProps) => css`
   margin: 0;
-  color: ${props.isSecondary ? cVar('colorCoreNeutral300') : cVar('colorCoreNeutral50')};
+  color: ${props.color ? props.color : props.isSecondary ? cVar('colorCoreNeutral300') : cVar('colorCoreNeutral50')};
```
Comment (drillprop) [inline src/components/Text/Text.styles.ts]: Let's refactor this a bit maybe? 
```js
const baseStyles = ({ color, isSecondary }: TextProps) => css`
  margin: 0;
  color: ${color ? color : isSecondary ? cVar('colorCoreNeutral300') : cVar('colorCoreNeutral50')};
`
```

**Review by drillprop — COMMENTED** [review]
Looks pretty solid in general

File: src/components/ListItem/ListItem.styles.ts (hunk 4 above)
Comment (drillprop) [reply src/components/ListItem/ListItem.styles.ts:29]: oh, wait, my mistake. I didn't understand the design. It's transparent by default, right?

File: src/components/ListItem/ListItem.styles.ts (hunk 5 above)
Comment (drillprop) [reply src/components/ListItem/ListItem.styles.ts:47]: https://github.com/Joystream/atlas/pull/1701#discussion_r761041470

File: src/components/ListItem/ListItem.styles.ts (hunk 4 above)
Comment (eldiegod) [reply src/components/ListItem/ListItem.styles.ts:29]: yep

**Comment by kdembler** [comment]
@eldiegod why are all the comments resolved? AFAICT there were no changes to the PR made after the comments were made so they shouldn't be resolved

**Comment by eldiegod** [comment]
> @eldiegod why are all the comments resolved? AFAICT there were no changes to the PR made after the comments were made so they shouldn't be resolved

@kdembler my bad, forgot to push them

**Comment by toiletgranny** [comment]
Omg, I can't believe this! This is so spot on! How did you do that?! You made my day, perhaps even week! There are some tiny-winy little details that I've found:

Here's all I found:
1. When the `destructive` property is set to `true`, the icon should inherit the red color as well. This is an important thing to have, I'm afraid.
2. When the `captionPosition` property is set to `right`, ideally the CSS `text-alignment` is also set to `right`, so that in rare scenarios when the text wraps, it stays aligned to the right side.
3. As I can't test this in Storybook (which is understandable), I just want to make sure: is it still possible to replace the icon with any other component we want in the future? Eg. an avatar?

**Comment by kdembler** [comment]
Just wanted to drop in and mention: great job on the story, I love that you took time to include all the arguments and all the different types of stories. I think that we could use this as an example on how the stories should look like 🥇

File: .env [hunk 7]
```diff
@@ -16,3 +16,5 @@ VITE_PRODUCTION_ORION_URL=https://orion.joystream.org/graphql
 VITE_PRODUCTION_NODE_URL=wss://rome-rpc-endpoint.joystream.org:9944
 VITE_PRODUCTION_FAUCET_URL=https://member-faucet.joystream.org/register
 VITE_PRODUCTION_OFFICIAL_JOYSTREAM_CHANNEL_ID=3
+VITE_PRODUCTION_HOTJAR_ID=2692611
+VITE_PRODUCTION_LIVESESSION_ID=029d5963.a21f20e4
```
Comment (drillprop) [inline .env]: These changes shouldn't be included

**Review by drillprop — CHANGES_REQUESTED** [review]
Looks good to me, but your PR still includes some weird commit. You need to rebase
image

**Review by drillprop — APPROVED** [review]

**Review by kdembler — APPROVED** [review]

**Pull request merged**